info-v4-commands.go

```diff
@@ -290,3 +297,96 @@ func (adm *AdminClient) DriveInfo(ctx context.Context, poolIndex, setIndex, disk
 
 	return disk, nil
 }
+
+//msgp:ignore NodeInfoOpts
+type NodeInfoOpts struct{}
+
+//msgp:ignore NodeListOpts
+type NodeListOpts struct {
+	Limit  int
+	Offset int
+}
+
+type NodeListResponse struct {
+	Nodes []Node `msg:"nodes,omitempty"`
+	Total int    `msg:"total,omitempty"`
+}
+
+// NodeList - list all nodes in the cluster
+func (adm *AdminClient) NodeList(ctx context.Context, options ...func(*NodeListOpts)) (nodeList *NodeListResponse, err error) {
```

Comment:
Maybe just...
```suggestion
func (adm *AdminClient) NodeList(ctx context.Context, options NodeListOpts) (nodeList *NodeListResponse, err error) {
```

.. or if you do prefer the functional approach...

```suggestion
func (adm *AdminClient) NodeList(ctx context.Context, options ...func(*NodeListOpts) error) (nodeList *NodeListResponse, err error) {
```

(and then also add the funcs to set limit/offset - though tbh it all seems kinda overkill)

Reply:
I just made this inline with what the other functions are doing. I don't mind either way tbh

Reply:
I think the ...func stuff was to make things more extendable without breaking consumers down the line

Reply:
Yeah - structs is usually fine and non-breaking... But if you want to do the func, it should be able to return an error.